src/librustc_resolve/lib.rs

```diff
@@ -99,16 +99,56 @@ use std::usize;
 use resolve_imports::{Target, ImportDirective, ImportResolution};
 use resolve_imports::Shadowable;
 
-
 // NB: This module needs to be declared first so diagnostics are
 // registered before they are used.
 pub mod diagnostics;
 
+macro_rules! resolve_err {
```

Comment:
In particular, if all of the places where you current write `resolve_err!` instead called into one centralized error method in `Resolver`, then *that* central error method would then be the place that says, at the beginning:
```rust
if !resolver.emit_errors { return; }
```